system/ee/legacy/core/Input.php

```diff
@@ -208,11 +208,37 @@ protected function _set_cookie(array $data)
 			$data['httponly'] = ( ! ee()->config->item('cookie_httponly')) ? 'y' : ee()->config->item('cookie_httponly');
 		}
 
-		//  Turn httponly into a true boolean.
-		$data['httponly'] = ($data['httponly'] == 'y' ? TRUE : FALSE);
+		// Turn httponly into a true boolean.
+		$data['httponly'] = get_bool_from_string($data['httponly']);
 
 		// Deal with secure cookies.
-		$data['secure_cookie'] = (bool_config_item('cookie_secure') === TRUE) ? 1 : 0;
+		$data['secure_cookie'] = bool_config_item('cookie_secure');
+
+		// SameSite explained: https://web.dev/samesite-cookies-explained/
+		// Allow default setting for all site cookies, then allow for overrides when this method is called.
+		// Cookies without a SameSite attribute will be treated as SameSite=Lax.
+		$sameSiteConfigValue = ee()->config->item('cookie_samesite');
```

Comment:
I'm honestly not sure what the behavior should be here. Should the config value set the site wide default setting of the SameSite cookie, or should it override _all_ cookie settings, regardless of what EE or a 3rd party add-on might want to set their cookies as? Part of me thinks this should change to just default all cookie to Lax, then move the cookie_samesite config override to the very end, to let site admins override settings to increase security, and potentially override what a 3rd party add-on, or even EE's own cookies are doing.

Reply:
I think the way you have it is fine. Default to lax or whatever level the site admin sets and then anyone setting a cookie can override it. As this option didn't exist until now, there shouldn't be any add-ons that accidentally set it to a value so any use of it going forward would be intentional on the dev's part and we should allow that. If anyone has any objections or concerns to this approach, please speak up.

Reply:
The reason I was thinking of moving the config option to the end is that if an add-on does end up setting a cookie to Lax, but then for security reasons, maybe the actual site maintainer wants to force everything to Strict.

We could take it a step further and add some sort of per cookie override. Either a hook every time a cookie is attempted to be set, or a config array to set the value for each array, e.g.

```
$config['cookie_samesite_overrides'] = [
    'publisher_site_language' => 'Strict',
    'untrusted_addon_cookie' => 'Strict',
    ...
];
```

Reply:
Oh wait, there is already a hook... `set_cookie_end`

Reply:
In that case, then I guess my current implementation is OK. If someone, a power user, or someone who want's to ensure all their cookies are perfectly set, can use the hook to change the value that a 3rd party add-on might set.

Reply:
We could add another config override to override any other cookies set in the hook:
`ee()->config->item('enforce_addon_cookie_samesite');` or some better name I can't think of at the moment.

Reply:
We discussed it internally and decided we should do it the way you suggested in your first comment:

- No config, not specified by add-on = default to LAX
- No config, specified by add-on = add-on setting
- Config, specified by add-on = config override (ignores add-on setting)

Would you be willing to update your PR to make the config override what is set by the add-on?